models/biocro/R/run.biocro.R

```diff
@@ -193,19 +252,19 @@ run.biocro <- function(lat, lon, metpath, soil.nc = NULL, config = config, coppi
   
   daily_grp <- dplyr::group_by_at(.tbl = hourly.results, .vars = "year")
   annual.results <- dplyr::bind_cols(
-      dplyr::summarize_at(
-        .tbl = daily_grp,
-        .vars = c("Stem", "Leaf", "Root", "AboveLitter", "BelowLitter",
-                  "Rhizome", "Grain"),
-        .fun = max),
-      dplyr::summarize_at(
-        .tbl = daily_grp,
-        .vars = c("SoilEvaporation", "CanopyTrans", map="precip"),
-        .fun = sum),
-      dplyr::summarize_at(
-        .tbl = daily_grp,
-        .vars = c(mat = "Temp"),
-        .fun = mean))
+    dplyr::summarize_at(
```

Comment:
@justinmcgrath can you think of other model outputs that will be of interest for model validation and prediction and should be available by default by the PEcAn workflow?

e.g. is there energy balance? soil respiration?

Reply:
Hopefully we'll have LAI data, which would be a perfect proxy for SLA in this case, so that would be good to include. We'll have height data, and that's somewhere in the model. I'll have to get it to output it. There is no energy balance or soil respiration.

I think I would write this differently though. Presumably whatever calls this function knows what it wants from the output, so it could pass the list of names as a parameter and this function could subset based on that, or this function could return all the values, and the caller could subset it.

Even those solutions aren't ideal. Currently, this violates single responsibility. The function runs the model and summarizes the result.

Reply:
@justinmcgrath 

* Do you mean compute LAI on the fly from leaf biomass * SLA?
* once you export height that would be great + easy to add

> Currently, this violates single responsibility. The function runs the model and summarizes the result.

You are correct; Typically PEcAn takes whatever a model's defaut output is and converts it to the PEcAn format in model2netcdf.<MODEL> functions. For BioCro I added this step for convenience: 1) so a rectangular .csv could be written and 2) to allow daily and annual summaries that are much smaller when runs are very large (regional + met + parameter uncertainty). At this point rather than refactoring I would drop the daily and annual summaries since PEcAn (currently) only uses the hourly output. Then if / when there is a need for daily or hourly summaries it should work on the PEcAn standard format and work for all models.

Reply:
LAI is one of the outputs. The name is 'lai'. It is currently calculated as leaf_mass * SLA, but I think it would be better to use the output directly.

Reply:
OK Looks like both Leaf biomass and LAI are included in the hourly output given to PEcAn (line 91 and 231 above )